pkg/cloud/services/ec2/securitygroups.go

```diff
@@ -379,7 +379,11 @@ func (s *Service) getSecurityGroupIngressRules(role infrav1.SecurityGroupRole) (
 				Protocol:    infrav1.SecurityGroupProtocolTCP,
 				FromPort:    6443,
 				ToPort:      6443,
-				CidrBlocks:  []string{anyIPv4CidrBlock},
+				SourceSecurityGroupIDs: []string{
+					s.scope.SecurityGroups()[infrav1.SecurityGroupAPIServerLB].ID,
+					s.scope.SecurityGroups()[infrav1.SecurityGroupControlPlane].ID,
+					s.scope.SecurityGroups()[infrav1.SecurityGroupNode].ID,
```

Comment:
We'll need `infrav1.SecurityGroupBastion` as well

Reply:
> We'll need `infrav1.SecurityGroupBastion` as well

I thought about that when creating the list of Security Groups. While we could add the bastion, I'm not we need it. The bastion can access the Load Balancer if you want to run `kubectl` from it. And you can SSH into the nodes if you want to try to connect directly to a local port.

This is really a question of whether the bastion is just there for outside access or if we want the bastion to serve a greater purpose like for troubleshooting. Thoughts?

Reply:
We can go with this for now and see if there is a need later